### PR TITLE
Oc 46 logout user after password reset

### PR DESCRIPTION
**Pull request by jeroendrenth**

### CODE REVIEW

**Comment by DrKlonk** [comment]
The logging out works nicely, but it does not fix OC-44. Whether we are logged in or not, if we keep the old window/tab open when resetting the password for a second time, the crash still occurs.

Maybe the question for OC-44 is if we want to spend time on such an abnormal use case.

**Comment by DrKlonk** [comment]
Actually, the logged in user that fills in their email to reset their password does not get logged out. This way, you can be logged in with 2 different passwords at the same time. I think that is unwanted behaviour. I see the issue OC-46 does not mention this.

File: frontend/pages/password/reset.vue [hunk 1]
```diff
@@ -88,6 +88,7 @@ export default defineComponent({
           password: newPassword1.value,
           passwordConfirmation: newPassword2.value,
         })
+        await $auth.logout()
```
Comment (DrKlonk) [inline frontend/pages/password/reset.vue:91]: As mentioned in a general comment, I think we should do this when we make the password reset request too.

File: frontend/pages/password/reset.test.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      newPassword1: 'welkombijordina01',
+      newPassword2: 'welkombijordina02',
+    })
+
+    expect(wrapper.vm.$data.passwordValidationError).toBe(
+      "The passwords don't match!",
+    )
+  })
+
+  test('Passwords and reset code are being send to strapi', async () => {
```
Comment (DrKlonk) [inline frontend/pages/password/reset.test.ts:81]: *sent

File: frontend/pages/password/reset.test.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
+import '~/test/jest.setup'
```
Comment (DrKlonk) [inline frontend/pages/password/reset.test.ts:1]: Very nice tests!

File: frontend/utils/helpers.ts [hunk 4]
```diff
@@ -1,8 +1,11 @@
...
-    return errorResponse.response.data.message[0].messages[0].message // WV 20210812 this is not fool-proof but probably good enough
+import { AxiosError } from 'axios'
+import { ErrorResponse } from './types'
+
+function errorMessageFromResponse(errorResponse: AxiosError<ErrorResponse>) {
+  if (Array.isArray(errorResponse.response))
+    return errorResponse?.response.data.message[0].messages[0].message // WV 20210812 this is not fool-proof but probably good enough
   return (
-    errorResponse.response.data.message ||
+    errorResponse.response?.data.message ||
```
Comment (DrKlonk) [inline frontend/utils/helpers.ts:8]: Yeah, better

**Comment by jeroendrenth** [comment]
I've added a middleware function. to check if a user is still logged in, even after he as logged out in another window or tab. This unfortunately does not work by just adding middleware: 'auth' to your component as described here: https://auth.nuxtjs.org/guide/middleware

**Comment by DrKlonk** [comment]
The merge of develop (and the fixing of the BaseContainer) really messed up the profile page. Could you add some containers to fix that?

File: frontend/utils/helpers.ts (hunk 4 above)
Comment (DrKlonk) [reply frontend/utils/helpers.ts:8]: Actually, this returns an array of objects when trying a wrong password.
image

Comment (DrKlonk) [reply frontend/utils/helpers.ts:8]: Which leads to an error in the frontend

File: frontend/pages/password/index.vue [hunk 5]
```diff
@@ -34,6 +34,7 @@
 import { defineComponent, useContext, ref } from '@nuxtjs/composition-api'
 
 export default defineComponent({
+  middleware: 'auth',
```
[the remark this answers is not in the document]
Comment (jeroendrenth) [reply frontend/pages/password/index.vue:37]: Why would you go to the reset password page when you're not logged in?

[the remark this answers is not in the document]
Comment (jeroendrenth) [reply frontend/pages/password/index.vue]: Removed it

[the remark this answers is not in the document]
Comment (DrKlonk) [reply frontend/pages/password/index.vue]: The "forgot password" use case, coming from the login page, also uses this page. So sometimes, you're not logged in coming to this page

**Comment by jeroendrenth** [comment]
Error handler aangepast

**Comment by jeroendrenth** [comment]
> The merge of develop (and the fixing of the BaseContainer) really messed up the profile page. Could you add some containers to fix that?

Done. Please check if it's done in an ok way

**Review by DrKlonk — APPROVED** [review]
Top!

**Pull request merged**